department-of-veterans-affairs/caseflow
# Creates IDT auth framework

Connects #6254

### Description
Adds a prototype of authentication functionality for the IDT to make authenticated API requests to Caseflow. No DB data is returned at this point, so we don't need to feature flag anything.

### Test plan
1. `curl localhost:3000/idt/api/v1/token`
1. Verify response is like `{ one_time_key: "dfsafjslakfs", token: "fdsakfjsa"}`
1. `curl localhost:3000/idt/api/v1/appeals`
1. Verify 400 is returned
1. `curl localhost:3000/idt/api/v1/appeals -H "token: fdsakfjsa"`
1. Verify 403 is returned
1. Open `localhost:3000/idt/auth?one_time_key=dfsafjslakfs`
1. Verify success message is shown
1. Open `localhost:3000/idt/auth?one_time_key=dfsafjslakfs`
1. Verify error is shown
1. `curl  -H "TOKEN: fdsakfjsa" localhost:3000/idt/api/v1/appeals`
1. Verify 200 is returned
